resources/splunk_app.rb

```diff
@@ -136,3 +148,41 @@ def config_scope
     apply_config
   end
 end
+
+class PackagedApp < SplunkApp
+  property :source_url, String, required: true
+
+  resource_name :splunk_app_package
+
+  def package_path
+    return @package_path if @package_path
+    file_cache = Pathname.new(Chef::Config['file_cache_path'])
+    @package_path = file_cache.join(CernerSplunk::PathHelpers.filename_from_url(source_url).gsub(/.spl$/, '.tgz'))
+  end
+
+  action :install do
+    return unless !version || changed?(:version)
+
+    remote_file package_path.to_s do
```

Comment:
So, because I don't have a mechanism currently to inspect the tarball, I will need to extract it to a temporary directory before installing the packaged app. Because of this I can delete local content instead of failing outright. Do you think this would be preferable @acharlieh ?